x-pack/plugins/spaces/index.ts

```diff
@@ -180,5 +181,21 @@ export const spaces = (kibana: Record<string, any>) =>
 
       // Register a function with server to manage the collection of usage stats
       server.usage.collectorSet.register(getSpacesUsageCollector(server));
+
+      server.registerCapabilitiesProvider(async (request, uiCapabilities) => {
+        const spacesClient = server.plugins.spaces.spacesClient.getScopedClient(request);
+        try {
```

Comment:
Is there a possibility that if something throws here, that by returning `uiCapabilities`, we accidentally leak a capability to the user that shouldn't?

Reply:
I need some guidance from @elastic/kibana-security here. In `replaceInjectedVars` on the server, if we can't load the active space, we do not modify `uiCapabilities`, but we do mark the `activeSpace` as invalid. What should we do in this case where the frontend is requesting the capabilities? 

This is only fetched when the UI code is first bootstrapped, so this case may not ever happen right now if we're attempting to hit an endpoint for an invalid space.

Reply:
At the moment, we don't remove entries from the ui capabilities data-structure, we only set their value to `false` if the user doesn't have access, so I'm not too worried about us leaking capabilities given the current implementation, but you do bring up an interesting point that I'll ponder further.